oak_functions/loader/tests/integration_test.rs

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use futures::executor::block_on;
+use maplit::hashmap;
+use oak_functions_abi::proto::Request;
+use oak_functions_loader::{
+    logger::Logger,
+    lookup::{LookupData, LookupDataSource},
+    server::WasmHandler,
+};
+use once_cell::sync::Lazy;
+use std::{io::Write, sync::Arc};
+use tokio;
+
+static WASM_MODULE_BYTES: Lazy<Vec<u8>> = Lazy::new(|| {
+    let mut manifest_path = std::env::current_dir().unwrap();
+    manifest_path.push("tests");
+    manifest_path.push("module");
+    manifest_path.push("Cargo.toml");
+
+    test_utils::compile_rust_wasm(manifest_path.to_str().unwrap(), false)
+        .expect("Could not read Wasm module")
+});
+
+#[tokio::test]
+async fn test_consistent_lookup() {
+    let entries = hashmap! {
+       b"ConsistentLookup".to_vec() => b"Value1".to_vec(),
+    };
+    let serialized_entries = test_utils::serialize_entries(entries);
+    let temp_file = tempfile::NamedTempFile::new().unwrap();
```

Comment:
Some of this logic could probably be encapsulated in a helper fn at some point. I also thought we have a way of setting the data for test without going through a file, but I'm not sure